src/wdb/QueryRunner.cc

```diff
@@ -333,6 +337,12 @@ bool QueryRunner::_dispatchChannel() {
             // metadata store of the worker, or keeping the state transisitons in a separate transient
             // store that won't be affected by the task destruction.
             _task->finished(std::chrono::system_clock::now());
+#endif
+            // DM-41358: Unlike the above-made call to _task->finished() that was disabled,
+            //           this one does the complete cleanup of metadata and stats associated
+            //           with the task completion. In the previous version of the code this
+            //           operation was initiated by BlendScheduler::commandFinish.
+            _queriesAndChunks->finishedTask(_task);
```

Comment:
This should probably go in the `class Release` defined in `runQuery(...)` so it's called even if there's an exception.
Also, `startedTask()` should probably be moved out of `BlendScheduler::commandStart()` and put just before `class Release` is instantiated so that it's very unlikely one gets called without the other.